### PR TITLE
[luci/pass] Introduce GeluPatternBase

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR introduces GeluPatternBase class.

for issue: https://github.com/Samsung/ONE/issues/11002
from draft: https://github.com/Samsung/ONE/pull/11016

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 1]
```diff
@@ -62,10 +89,10 @@ bool same(float a, float b) { return fabs(a - b) < 1e-5; }
  *          [Out]
  *
  */
-class GeluPattern final
+class GeluPattern final : public GeluPatternBase
 {
 public:
-  GeluPattern(luci::CircleMul *candidate)
+  GeluPattern(luci::CircleMul *candidate) : GeluPatternBase(candidate)
   {
```
Comment (BalyshevArtem) [inline compiler/luci/pass/src/FuseGeluPass.cpp:96]: just to avoid `define but not used` error

**Comment by seanshpark** [comment]
it would be better to update #11016 and use this as draft and drop #11020

File: compiler/luci/pass/src/FuseGeluPass.cpp [hunk 2]
```diff
@@ -91,6 +118,51 @@ class GeluPattern final
   if (not(condition))             \
     return false;
 
+bool GeluPatternBase::matched_base()
```
[the remark this answers is not in the document]
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp:121]: okay, I will remove this

[the remark this answers is not in the document]
Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp]: Done

**Comment by BalyshevArtem** [comment]
> it would be better to update #11016 and use this as draft and drop #11020

I'm not sure I understood you correctly. If I understand correctly, do you want me to update one of these drafts with the content from this PR? I would like the overall picture to remain visible, from where the current PR stands out, and therefore leave them, but if it will bettter to remove them, then I will.

**Comment by seanshpark** [comment]
> do you want me to update one of these drafts

I mean, would be better to use (link in the message) first draft not #11020

File: compiler/luci/pass/src/FuseGeluPass.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/pass/src/FuseGeluPass.cpp:93]: can you check this as the method is now removed?

Comment (BalyshevArtem) [reply compiler/luci/pass/src/FuseGeluPass.cpp:93]: I checked, without this CI crashes with an error `define but not used`

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**